### PR TITLE
add export_bgen

### PR DESCRIPTION
**Pull request by cseed**
Exports BGEN 1.2 with 8 bits per probability.

Needs docs and more tests, not quite ready yet.

### CODE REVIEW

File: hail/python/test/hail/methods/test_impex.py [hunk 1]
```diff
@@ -1088,6 +1088,17 @@ def test_specify_different_index_file(self):
...
             hl.index_bgen(bgen_file, index_file_map=index_file_map)
 
+    def test_bgen_export_same(self):
+        bgen = hl.import_bgen(resource('example.8bits.bgen'),
+                              entry_fields=['GP'],
+                              sample_file=resource('example.sample'))
+        tmp = new_temp_file()
+        hl.export_bgen(bgen, tmp)
+        hl.index_bgen(tmp + '.bgen')
+        self.assertTrue(bgen._same(hl.import_bgen(tmp + '.bgen',
```
Comment (tpoterba) [inline hail/python/test/hail/methods/test_impex.py]: prefer plain python `assert` here -- pytest error messages are better than unittest's

File: hail/src/main/scala/is/hail/io/gen/ExportBGEN.scala [hunk 2]
```diff
@@ -0,0 +1,371 @@
...
+          gpSum += x
+          gpResized(idx) = x * totalProb // Assuming sum(GP) == 1
+          idx += 1
+        }
+
+        if (gpSum >= 0.999 && gpSum <= 1.001) {
+          uncompressedData(samplePloidyStart + i) = ploidy
+          roundWithConstantSum(gpResized, fractional, index, indexInverse, uncompressedData, totalProb)
+        } else {
+          warn(s"GP sum was not in the range [0.999, 1.001]. Found $gpSum. Emitting missing probabilities.")
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/io/gen/ExportBGEN.scala]: this warning will go to the worker logs, creating massive logs if something is wrong in the data but not propagating back to the user. Prefer an incremented count of bad GPs per partition which is collected with the partition file name.

**Comment by cseed** [comment]
Thanks, @tpoterba!  I wasn't done so I hadn't assigned anyone, just wanted to run the tests, but it should be good to go now.

**Comment by cseed** [comment]
Note, @jigold gets the credit for this, ExportBGEN.scala was basically taken from an old PR of hers that never made it in.  I just simplified it to only support the 8-bit case.

**Pull request merged**